### PR TITLE
Fix anti-snowball

### PR DESCRIPTION
**Pull request by komuta**
* fix the issue leading to instances being killed too quickly
* fix the perf issue by making the whole code asynchroneous

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala [hunk 1]
```diff
@@ -159,32 +160,24 @@ private[health] class HealthCheckActor(
...
-                case Success(activeInstances) => logger.info(s"Done killing $instanceId, $activeInstances remaining")
-                case Failure(throwable) => logger.warn(s"[anti-snowball] Error when counting instance remaining: ${throwable.getMessage}. Since it is just for logging purpose, we don't care that much")
-              }
-            case Failure(t) => logger.error(s"An error has occurred: ${t.getMessage}", t)
-          }
-          case Failure(_) => logger.error(s"Timed out killing instance $instanceId")
-        }
+        // TODO prevent killingInFlight from growing by cleaning killed instances (probably in purgeStatusOfDoneInstances)
+        killingInFlight = killingInFlight + instanceId
+        logger.debug(s"[anti-snowball] killing ${instanceId}, killingInFlight = ${killingInFlight}")
```
Comment (kamaradclimber) [inline src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala]: Are we sure logging a potentially large set is a good idea?

**Review by kamaradclimber — APPROVED** [review]

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala (hunk 1 above)
Comment (komuta) [reply src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala]: No, this was done for debugging purpose, and should be removed :)

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala [hunk 2]
```diff
@@ -162,7 +165,7 @@ private[health] class HealthCheckActor(
         )
         // TODO prevent killingInFlight from growing by cleaning killed instances (probably in purgeStatusOfDoneInstances)
```
Comment (kamaradclimber) [inline src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala]: you should remove this line

**Review by kamaradclimber — APPROVED** [review]

**Review by kamaradclimber — APPROVED** [review]

File: src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala [hunk 3]
```diff
@@ -170,12 +169,14 @@ private[health] class HealthCheckActor(
     }
   }
 
+  /** Check if enough active and ready instances will remain if we kill 1 unhealthy instance */
```
Comment (kamaradclimber) [inline src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala:172]: good idea to add this comment to summarize the high level goal :)

**Review by kamaradclimber — APPROVED** [review]

**Pull request merged**